OpenRA.Platforms.Default/ThreadedGraphicsContext.cs

```diff
@@ -168,7 +154,9 @@ public Message(ThreadedGraphicsContext device)
 			volatile Action<object> actionWithParam;
 			volatile Func<object> func;
 			volatile Func<object, object> funcWithParam;
+			volatile Func<object, Type, object> funcWithParamAndGeneric;
```

Comment:
Don't add a new overload for this. You can use the existing `Func<object, object> funcWithParam;` overload. Pack all the parameters into a tuple which can be passed as an object, then unpack the parameters on the other side.

This avoids us having to create an explosion of delegates to handle every variation of possible parameters.